src/xo-file.c

```diff
@@ -978,7 +978,8 @@ void xoj_parser_text(GMarkupParseContext *context,
     /* Note: comparing floats with "==" is dangerous, though in the case of dummy points, both floats
        will have been computed in the same way from the same ascii string, so they should really be
        absolutely equal. */
-    if (ui.cur_path.coords[0] == ui.cur_path.coords[2] && ui.cur_path.coords[1] == ui.cur_path.coords[3]) {
+    if (ui.cur_path.num_points >= 3 &&
```

Comment:
If the number of points is 2 and the points are different then you drop the first point? even if they are different?